modules/extrinsic-restrict/src/lib.rs

```diff
@@ -0,0 +1,336 @@
+// This file is part of Acala.
+
+// Copyright (C) 2020-2022 Acala Foundation.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Extrinsic Restrict module.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+use frame_support::{
+	log,
+	pallet_prelude::*,
+	traits::{Contains, IsSubType},
+	transactional, PalletId,
+};
+use frame_system::{ensure_signed, pallet_prelude::*};
+use module_support::{Rate, Ratio};
+use orml_traits::Change;
+use orml_traits::MultiCurrency;
+use primitives::{Balance, CurrencyId};
+use scale_info::TypeInfo;
+use sp_arithmetic::traits::CheckedRem;
+use sp_runtime::{
+	traits::{
+		AccountIdConversion, BlockNumberProvider, Bounded, CheckedDiv, CheckedSub, Convert, DispatchInfoOf, One,
+		PostDispatchInfoOf, Saturating, SignedExtension, UniqueSaturatedInto, Zero,
+	},
+	transaction_validity::{InvalidTransaction, TransactionValidityError},
+	ArithmeticError, FixedPointNumber,
+};
+use sp_std::{cmp::Ordering, convert::From, marker::PhantomData, prelude::*, vec, vec::Vec};
+use xcm::latest::prelude::*;
+use xcm::VersionedMultiAsset;
+
+pub use module::*;
+// pub use weights::WeightInfo;
+
+mod mock;
+mod tests;
+pub mod weights;
+
+#[frame_support::pallet]
+pub mod module {
+	use super::*;
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, Default)]
+	pub struct XtokensTransferLimit {
```

Comment:
```suggestion
	pub struct RateLimit {
```

no need to reference xtokens in this pallet